pyvrp/cpp/CostEvaluator.h

```diff
@@ -23,6 +23,51 @@ template <typename T> concept CostEvaluatable = requires(T arg)
     // clang-format on
 };
 
+/**
+ * RouteData(
+ *     size: int,
+ *     distance: Distance,
+ *     load: Load,
+ *     timeWarp: Duration
+ * )
```

Comment:
I am not sure about the name. 

We use this exclusively in `pyvrp.search`, so putting this in `CostEvaluator` feels like the wrong place. Perhaps a new file in `pyvrp/cpp/search`?

Since this "route data" object essentially just holds the properties related to cost that a route will have after a proposed move is applied, a name like `MoveProperties` seems clearer.